OpenTap.Plugins.PNAX/Converters/Modulation Distortion/MODXMixer.cs

```diff
@@ -0,0 +1,295 @@
+            retVal.Add(("LO1MixerFrequency", LO1MixerFrequency));
+            if (ConverterStages == ConverterStagesEnum._2)
+            {
+                retVal.Add(("IFMixerFrequency", IFMixerFrequency));
+                retVal.Add(("LO2FractionalMultiplierNumerator", LO2FractionalMultiplierNumerator));
+                retVal.Add(("LO2FractionalMultiplierDenominator", LO2FractionalMultiplierDenominator));
+                retVal.Add(("LO2MixerFrequency", LO2MixerFrequency));
+            }
+
+            retVal.Add(("EnableEmbeddedLO", EnableEmbeddedLO));
+            if (EnableEmbeddedLO)
+            {
+                retVal.Add(("TuningMethod", TuningMethod));
+                retVal.Add(("TuneEvery", TuneEvery));
+                retVal.Add(("BroadBandSearch", BroadBandSearch));
+                retVal.Add(("NoiseBW", NoiseBW));
+                retVal.Add(("MaxIterations", MaxIterations));
+                retVal.Add(("Tolerance", Tolerance));
+                retVal.Add(("LOFrequencyDelta", LOFrequencyDelta));
+            }
+            retVal.Add(("LO1Source", LO1Source));
+            retVal.Add(("LO1Power", LO1Power));
+            retVal.Add(("LO1SourceLevelingMode", LO1SourceLevelingMode));
+            retVal.Add(("LO1Attenuator", LO1Attenuator));
+
+            if (ConverterStages == ConverterStagesEnum._2)
+            {
+                retVal.Add(("LO2Source", LO2Source));
+                retVal.Add(("LO2Power", LO2Power));
+                retVal.Add(("LO2SourceLevelingMode", LO2SourceLevelingMode));
+                retVal.Add(("LO2Attenuator", LO2Attenuator));
+            }
+            return retVal;
+        }
+
+        public override void Run()
+        {
+            RunChildSteps(); //If the step supports child steps.
+
+            // Start from scratch
+            PNAX.MixerDiscard(Channel);
+
+            // Setup
+            PNAX.SetConverterStages(Channel, ConverterStages);
+            PNAX.MODCarrierFreq(Channel, InputMixerFrequency);
+            PNAX.SetFrequencyOutputSideband(Channel, OutputMixerFrequency);
+
+            PNAX.SetLOFractionalMultiplierNumerator(Channel, 1, LO1FractionalMultiplierNumerator);
+            PNAX.SetLOFractionalMultiplierDenominator(Channel, 1, LO1FractionalMultiplierDenominator);
+            PNAX.SetFrequencyLOFixed(Channel, 1, LO1MixerFrequency);
+            if (ConverterStages == ConverterStagesEnum._2)
+            {
+                PNAX.SetFrequencyIFSideband(Channel, IFMixerFrequency);
+
+                PNAX.SetLOFractionalMultiplierNumerator(Channel, 2, LO2FractionalMultiplierNumerator);
+                PNAX.SetLOFractionalMultiplierDenominator(Channel, 2, LO2FractionalMultiplierDenominator);
+                PNAX.SetFrequencyLOFixed(Channel, 2, LO2MixerFrequency);
+                PNAX.MixerCalc(Channel);
+            }
+            //PNAX.MixerApply(Channel);
```

Comment:
I didn't test with Mixer, I am not sure if the Apply needs to be done at every state, that is why I left that commented line there.

there is also another command PNAX.MODMixerSourceRole that is not explicitly called in the PNA UI, but the help file indicated that the command was available for Mixers. I left it commented in case its needed in the future.